[ui5-tooling-transpile] Exception produced during plugin execution when babel.config contains plugins

**Describe the bug**
There is an issue in `ui5-tooling-transpile` version 3.0.4 (not present in 3.0.3) when used with a `babel.config` that includes plugin definitions.

Given this `babel.config`:
```
{
  "ignore": [
    "**/*.d.ts"
  ],
  "plugins": [
    "@babel/plugin-transform-strict-mode"
  ],
  "presets": [
    "transform-ui5",
    [
      "@babel/preset-typescript",
      {
        "allowDeclareFields": true
      }
    ]
  ],
  "sourceMaps": true
}
```

We are getting the following error:

```
[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] /data/jenkins/workspace/anch_development_ui-apps_PR-2106/node_modules/karma-ui5-transpile/lib/index.js:74

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] 				return plugin.file.request === "istanbul";

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] 				              ^

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] TypeError: Cannot read properties of undefined (reading 'file')

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ]     at /data/jenkins/workspace/anch_development_ui-apps_PR-2106/node_modules/karma-ui5-transpile/lib/index.js:74:19

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ]     at Array.find (<anonymous>)

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ]     at /data/jenkins/workspace/anch_development_ui-apps_PR-2106/node_modules/karma-ui5-transpile/lib/index.js:73:26

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] Node.js v18.16.0

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ] /data/jenkins/workspace/anch_development_ui-apps_PR-2106/node_modules/karma-ui5-transpile/lib/index.js:74

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ] 				return plugin.file.request === "istanbul";

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ] 				              ^

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ] 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ] TypeError: Cannot read properties of undefined (reading 'file')

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ]     at /data/jenkins/workspace/anch_development_ui-apps_PR-2106/node_modules/karma-ui5-transpile/lib/index.js:74:19

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ]     at Array.find (<anonymous>)

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ]     at /data/jenkins/workspace/anch_development_ui-apps_PR-2106/node_modules/karma-ui5-transpile/lib/index.js:73:26

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ] 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ] Node.js v18.16.0

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] npm ERR! Lifecycle script `karma-cov-ci` failed with error: 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] npm ERR! Error: command failed 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] npm ERR!   in workspace: configurator@3.14.16 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] npm ERR!   at location: /data/jenkins/workspace/anch_development_ui-apps_PR-2106/configurator 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ] npm ERR! Lifecycle script `karma-cov-ci` failed with error: 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ] npm ERR! Error: command failed 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ] npm ERR!   in workspace: sap.via.shared@3.14.16 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:sap.via.shared         ] npm ERR!   at location: /data/jenkins/workspace/anch_development_ui-apps_PR-2106/sap.via.shared 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] 

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] > sap.via.applications@3.14.16 karma-cov-ci:module:configurator

[2023-08-25T03:42:12.321Z] [karma-cov-ci:module:configurator           ] > npm run karma-cov-ci -w configurator
```

The following code in `babel-core` is creating a babel config with `plugins` and `presets` fields that have a type of  `ConfigItem[]`:

https://github.com/babel/babel/blob/e43d8e77cf88245e930055e70a42f46e2996bce6/packages/babel-core/src/config/partial.ts#L119

https://github.com/babel/babel/blob/e43d8e77cf88245e930055e70a42f46e2996bce6/packages/babel-core/src/config/item.ts#L9

When `normalizePresetOrPlugin` is passed a `ConfigItem` in the `babelPresetOrPlugin` argument, it returns the `ConfigItem` object that was passed in (in other cases a string identifying a module that can be passed into `require.resolve()` is returned):
https://github.com/ui5-community/ui5-ecosystem-showcase/blob/9127d611a58bd7677bff96480f652f4eeea53172/packages/ui5-tooling-transpile/lib/util.js#L116

This  results in a `ConfigItem` being passed as the `moduleName` argument to `resolveNodeModule`: 
https://github.com/ui5-community/ui5-ecosystem-showcase/blob/9127d611a58bd7677bff96480f652f4eeea53172/packages/ui5-tooling-transpile/lib/util.js#L24C10-L24C27

Both invokations of `require.resolve()` return `undefined` as the `request` argument is expected to contain a string rather than an object, meaning `resolveNodeModule` returns `undefined`.

This results in `babelOptions.plugins` containing an array of length 1 containing an `undefined` value, which causes the exception at:
https://github.com/ui5-community/ui5-ecosystem-showcase/blob/9127d611a58bd7677bff96480f652f4eeea53172/packages/karma-ui5-transpile/lib/index.js#L73

**To Reproduce**

Use a babel.config with some plugins defined (see above) with the 
ui5-tooling-transpile plugin.

The issue was occurring when running tests using a `package.json` containing the following `devDepenendencies`:

```jsonc
  "devDependencies": {
    "@babel/plugin-transform-strict-mode": "7.22.5",
    "@sapui5/types": "1.116.0",
    "@types/web": "0.0.114",
    "@ui5/builder": "3.0.9",
    "@ui5/cli": "3.4.0",
    "@ui5/fs": "3.0.4",
    "@ui5/ts-interface-generator": "0.7.0",
    "adm-zip": "0.5.10",
    "http-proxy-middleware": "2.0.6",
    "jsonc-parser": "3.2.0",
    "karma": "6.4.2",
    "karma-chrome-launcher": "3.2.0",
    "karma-coverage": "2.2.1",
    "karma-junit-reporter": "2.0.1",
    "karma-sinon": "1.0.5",
    "karma-sonarqube-generic-reporter": "1.0.0",
    "karma-sourcemap-loader": "0.4.0",
    "karma-ui5": "3.0.3",
    "karma-ui5-transpile": "3.1.1",
    "local-web-server": "5.3.0",
    "lodash.merge": "4.6.2",
    "minimatch-all": "1.1.0",
    "mvn-artifact-download": "6.1.1",
    "npm-run-all": "4.1.5",
    "rimraf": "5.0.1",
    "sinon": "15.2.0",
    "ts-standard": "12.0.2",
    "typescript": "5.1.6"
  },
```

The issue does not occur when `"ui5-tooling-transpile": "3.0.3",` is added.

This has been observed when running karma tests on a UI5 project using TypeScript.
I can provide a link to the actual project (accessible within SAP) if there are issues with reproducing the issue.

**Expected behavior**

The observed exception should not occur as the `babel.config` file is valid according to the babel documentation.

**Environment**

This is not related to the OS, node version etc.

--- a/packages/karma-ui5-transpile/lib/index.js
+++ b/packages/karma-ui5-transpile/lib/index.js
@@ -65,28 +65,43 @@ function createPreprocessor(config, logger) {
 	const configuration = createConfiguration(transpileTask?.configuration);
 
 	// create the Babel configuration using the ui5-tooling-tranpile-task util
-	const babelConfigCreated = createBabelConfig({ configuration, isMiddleware: false }).then((babelOptions) => {
-		// and inject the babel-plugin-istanbul into the configuration
-		// if not already configured in the plugins section
-		babelOptions.plugins = babelOptions.plugins || [];
-		if (
-			!babelOptions.plugins.find((plugin) => {
-				return plugin.file.request === "istanbul";
-			})
-		) {
-			const istanbulConfig = {
-				include: ["**/*"],
-				exclude: []
-			};
-			// apply the `instrumenterOptions` for istanbul from `coverageReporter` in the `karma.config`
-			const instrumenterOptionsIstanbul = config.coverageReporter?.instrumenterOptions?.istanbul;
-			if (typeof instrumenterOptionsIstanbul === "object") {
-				Object.assign(istanbulConfig, instrumenterOptionsIstanbul);
+	const babelConfigCreated = createBabelConfig({
+		configuration,
+		isMiddleware: false,
+		preprocess: function (babelOptions) {
+			// and inject the babel-plugin-istanbul into the configuration
+			// if not already configured in the plugins section
+			babelOptions.plugins = babelOptions.plugins || [];
+			if (
+				!babelOptions.plugins.find((plugin) => {
+					let pluginName;
+					if (typeof plugin === "string") {
+						// plugin defined as string only
+						pluginName = plugin;
+					} else if (Array.isArray(plugin)) {
+						// plugin defined with config
+						pluginName = plugin[0];
+					} else if (plugin.file) {
+						// plugin already processed by babel
+						pluginName = plugin.file.request;
+					}
+					return pluginName === "istanbul";
+				})
+			) {
+				// default include/exclude configuration for istanbul
+				const istanbulConfig = {
+					include: ["**/*"],
+					exclude: []
+				};
+				// apply the `instrumenterOptions` for istanbul from `coverageReporter` in the `karma.config`
+				const instrumenterOptionsIstanbul = config.coverageReporter?.instrumenterOptions?.istanbul;
+				if (typeof instrumenterOptionsIstanbul === "object") {
+					Object.assign(istanbulConfig, instrumenterOptionsIstanbul);
+				}
+				// add istanbul as first plugin into the plugins chain
+				babelOptions.plugins.unshift(["istanbul", istanbulConfig]);
 			}
-			// add istanbul as first plugin into the plugins chain
-			babelOptions.plugins.unshift(["istanbul", istanbulConfig]);
 		}
-		return babelOptions;
 	});
 
 	/**

--- a/packages/karma-ui5-transpile/package.json
+++ b/packages/karma-ui5-transpile/package.json
@@ -33,6 +33,6 @@
   },
   "peerDependencies": {
     "karma": ">=6.4.1",
-    "ui5-tooling-transpile": ">=3.0.0"
+    "ui5-tooling-transpile": ">=3.2.0"
   }
 }

--- a/packages/ui5-tooling-transpile/lib/util.js
+++ b/packages/ui5-tooling-transpile/lib/util.js
@@ -51,27 +51,37 @@ const CFG_FILES = [...multiply(".babelrc", [".json", ".js", ".cjs", ".mjs"]), ".
 
 // helper to load the babel configuration
 // eslint-disable-next-line jsdoc/require-jsdoc
-async function loadBabelConfig(babelConfig, skipBabelPresetPluginResolve, cwd = process.cwd()) {
+async function loadBabelConfigOptions(babelConfigOptions, skipBabelPresetPluginResolve, cwd = process.cwd()) {
 	// make the paths of the babel plugins and presets absolute
+	let fileToPresetOrPlugin;
 	if (!skipBabelPresetPluginResolve) {
-		babelConfig = normalizeBabelConfig(babelConfig, cwd);
+		const normalizationInfo = normalizeBabelConfigOptions(babelConfigOptions, cwd);
+		babelConfigOptions = normalizationInfo?.babelConfigOptions;
+		fileToPresetOrPlugin = normalizationInfo?.fileToPresetOrPlugin;
 	}
 	// let babel load the babel config
-	return babel.loadPartialConfigAsync(
+	const partialConfig = await babel.loadPartialConfigAsync(
 		Object.assign(
 			{},
 			{
 				configFile: false,
 				babelrc: false
 			},
-			babelConfig
+			babelConfigOptions
 		)
 	);
+	const loadedBabelConfigOptions = partialConfig?.options;
+	// correct the request property again
+	if (!skipBabelPresetPluginResolve) {
+		postprocessBabelConfigOptions(loadedBabelConfigOptions, fileToPresetOrPlugin);
+	}
+	// ready
+	return loadedBabelConfigOptions;
 }
 
 // helper to find the babel configuration
 // eslint-disable-next-line jsdoc/require-jsdoc
-async function findBabelConfig(cwd) {
+async function findBabelConfigOptions(cwd) {
 	let configFile;
 
 	const findConfigFile = function (cfgFiles, dir) {
@@ -90,24 +100,24 @@ async function findBabelConfig(cwd) {
 		cwd = path.dirname(cwd);
 	}
 
-	let babelConfig;
+	let babelConfigOptions;
 	if (path.basename(configFile) === "package.json") {
 		// for the package.json, we need to extract the babel config
 		const pkgJson = JSON.parse(fs.readFileSync(configFile, { encoding: "utf8" }));
 		if (pkgJson.babel) {
-			babelConfig = pkgJson.babel;
+			babelConfigOptions = pkgJson.babel;
 		} else {
-			babelConfig = configFile = undefined;
+			babelConfigOptions = configFile = undefined;
 		}
 	} else if (configFile) {
 		// for a babel config file we load it on our own to normalize the plugin/preset paths
 		// => no recursive merging of babel config is possible with this approach
-		babelConfig = JSONC.parse(fs.readFileSync(configFile, { encoding: "utf8" }));
+		babelConfigOptions = JSONC.parse(fs.readFileSync(configFile, { encoding: "utf8" }));
 		// let Babel lookup the configuration file with the Babel API
-		//partialConfig = await loadBabelConfig({ configFile, filename: dir, babelrc: true });
+		//partialConfig = await loadBabelConfigOptions({ configFile, filename: dir, babelrc: true });
 	}
 
-	return babelConfig ? { configFile, babelConfig } : undefined;
+	return babelConfigOptions ? { configFile, babelConfigOptions } : undefined;
 }
 
 // utility to normalize the name of the babel preset or plugin
@@ -146,31 +156,63 @@ function normalizePresetOrPlugin(babelPresetOrPlugin, isPreset) {
 
 // utility to normalize and resolve the babel preset or plugin
 // eslint-disable-next-line jsdoc/require-jsdoc
-function resolvePresetOrPlugin(babelPresetOrPlugin, isPreset, cwd = process.cwd()) {
+function resolvePresetOrPlugin(babelPresetOrPlugin, isPreset, fileToPresetOrPlugin = {}, cwd = process.cwd()) {
 	// either an array or string is processed here as ConfigItems
 	// should be ignored and just returned (already resolved by babel)
+	// => in case of the preset/plugin cannot be resolved, we return just
+	//    use the given preset/plugin name and let babel do the resolve
+	//    which may lead to issues in pnpm or global ui5 tooling scenarios (rare cases!)
+	// => in addition we add a hint to the fileTotPresetOrPlugin map so that later
+	//    the original name of the preset/plugin can be derived again
+	//    for the ConfigItem property: plugin|preset.file.request
 	if (Array.isArray(babelPresetOrPlugin)) {
+		const request = babelPresetOrPlugin[0];
 		const normalized = normalizePresetOrPlugin(babelPresetOrPlugin[0], isPreset);
-		babelPresetOrPlugin[0] = resolveNodeModule(normalized, cwd);
+		babelPresetOrPlugin[0] = resolveNodeModule(normalized, cwd) || babelPresetOrPlugin[0];
+		fileToPresetOrPlugin[babelPresetOrPlugin[0]] = request;
 	} else if (typeof babelPresetOrPlugin === "string") {
+		const request = babelPresetOrPlugin;
 		const normalized = normalizePresetOrPlugin(babelPresetOrPlugin, isPreset);
-		babelPresetOrPlugin = resolveNodeModule(normalized, cwd);
+		babelPresetOrPlugin = resolveNodeModule(normalized, cwd) || babelPresetOrPlugin;
+		fileToPresetOrPlugin[babelPresetOrPlugin] = request;
 	}
 	return babelPresetOrPlugin;
 }
 
 // helper to normalize and resolve the babel configuration (resolve plugins and presets to absolute paths)
 // eslint-disable-next-line jsdoc/require-jsdoc
-function normalizeBabelConfig(babelConfig, cwd = process.cwd()) {
+function normalizeBabelConfigOptions(babelConfigOptions, cwd = process.cwd()) {
+	const fileToPresetOrPlugin = {};
 	// resolve the presets
-	if (Array.isArray(babelConfig?.presets)) {
-		babelConfig.presets = babelConfig.presets.map((preset) => resolvePresetOrPlugin(preset, true, cwd));
+	if (Array.isArray(babelConfigOptions?.presets)) {
+		babelConfigOptions.presets = babelConfigOptions.presets.map((preset) =>
+			resolvePresetOrPlugin(preset, true, fileToPresetOrPlugin, cwd)
+		);
 	}
 	// resolve the plugins
-	if (Array.isArray(babelConfig?.plugins)) {
-		babelConfig.plugins = babelConfig.plugins.map((plugin) => resolvePresetOrPlugin(plugin, false, cwd));
+	if (Array.isArray(babelConfigOptions?.plugins)) {
+		babelConfigOptions.plugins = babelConfigOptions.plugins.map((plugin) =>
+			resolvePresetOrPlugin(plugin, false, fileToPresetOrPlugin, cwd)
+		);
 	}
-	return babelConfig;
+	return { babelConfigOptions, fileToPresetOrPlugin };
+}
+
+// helper to postprocess the babel configuration and restore the "request" property for the plugins/presets
+// eslint-disable-next-line jsdoc/require-jsdoc
+function postprocessBabelConfigOptions(loadedBabelConfigOptions, fileToPresetOrPlugin = {}) {
+	const fixFileRequest = function (loadedPresetsOrPlugins) {
+		if (Array.isArray(loadedPresetsOrPlugins)) {
+			loadedPresetsOrPlugins.forEach((presetOrPlugin) => {
+				const request = presetOrPlugin.file.request;
+				presetOrPlugin.file.request = fileToPresetOrPlugin[request] || request;
+			});
+		}
+	};
+	// fix the presets
+	fixFileRequest(loadedBabelConfigOptions?.presets);
+	// fix the plugins
+	fixFileRequest(loadedBabelConfigOptions?.plugins);
 }
 
 module.exports = function (log) {
@@ -254,7 +296,7 @@ module.exports = function (log) {
 			// return the normalized configuration object
 			const normalizedConfiguration = {
 				debug: config.debug,
-				babelConfig: config.babelConfig,
+				babelConfigOptions: config.babelConfigOptions,
 				includes,
 				excludes,
 				filePattern,
@@ -283,10 +325,15 @@ module.exports = function (log) {
 		 * @param {object} cfg configuration object
 		 * @param {object} cfg.configuration task/middleware configuration
 		 * @param {boolean} cfg.isMiddleware true, if the function is called from the middleware
+		 * @param {function} [cfg.preprocess] function to preprocess the babel configuration (before loaded)
+		 * @param {function} [cfg.postprocess] function to postprocess the babel configuration (after loaded)
 		 * @param {string} [cwd] the cwd to lookup the configuration (defaults to process.cwd())
 		 * @returns {object} the babel plugins configuration
 		 */
-		createBabelConfig: async function createBabelConfig({ configuration, isMiddleware }, cwd = process.cwd()) {
+		createBabelConfig: async function createBabelConfig(
+			{ configuration, isMiddleware, preprocess, postprocess },
+			cwd = process.cwd()
+		) {
 			// Things to consider:
 			//   - middleware uses configs from app also for dependencies
 			//   - task must provide the cwd from outside
@@ -307,50 +354,57 @@ module.exports = function (log) {
 			};
 
 			// utility to update the babel config
-			const updateBabelConfig = async function (babelConfig) {
+			const updateBabelConfigOptions = async function (babelConfigOptions) {
 				// in the middleware case we generate the sourcemaps inline for
 				// debugging purposes since the middleware may not know about the
 				// sourcemaps files next to the source file
 				if (isMiddleware) {
-					babelConfig.sourceMaps = "inline";
+					babelConfigOptions.sourceMaps = "inline";
+				}
+				// call the preprocess hook
+				if (typeof preprocess === "function") {
+					preprocess(babelConfigOptions);
 				}
 				// finally load the babel config
-				const partialConfig = await loadBabelConfig(
-					babelConfig,
+				const loadedBabelConfigOptions = await loadBabelConfigOptions(
+					babelConfigOptions,
 					configuration.skipBabelPresetPluginResolve,
 					cwd
 				);
-				const loadedBabelConfig = partialConfig?.options;
+				// call the postprocess hook
+				if (typeof postprocess === "function") {
+					postprocess(babelConfigOptions);
+				}
 				// some logging
-				configuration?.debug && log.verbose(`${JSON.stringify(loadedBabelConfig, null, 2)}`);
-				return loadedBabelConfig;
+				configuration?.debug && log.verbose(`${JSON.stringify(loadedBabelConfigOptions, null, 2)}`);
+				return loadedBabelConfigOptions;
 			};
 
 			// the inline babel configuration in the ui5.yaml wins
-			let babelConfig = configuration?.babelConfig;
-			if (babelConfig) {
+			let babelConfigOptions = configuration?.babelConfig;
+			if (babelConfigOptions) {
 				if (configuration?.debug) {
 					log.info(`Using inline Babel configuration from ui5.yaml...`);
 					warnAboutIgnoredConfig();
 				}
-				return await updateBabelConfig(babelConfig);
+				return await updateBabelConfigOptions(babelConfigOptions);
 			}
 
 			// lookup the babel config by file
-			const config = await findBabelConfig(cwd);
+			const config = await findBabelConfigOptions(cwd);
 			if (config) {
 				if (configuration?.debug) {
 					log.info(`Using Babel configuration from ${config.configFile}...`);
 					warnAboutIgnoredConfig();
 				}
-				return await updateBabelConfig(config.babelConfig);
+				return await updateBabelConfigOptions(config.babelConfigOptions);
 			}
 
 			// create configuration based on ui5.yaml configuration options
 			configuration?.debug && log.info(`Create Babel configuration based on ui5.yaml configuration options...`);
 
 			// create the babel configuration based on the ui5.yaml
-			babelConfig = { plugins: [], presets: [] };
+			babelConfigOptions = { plugins: [], presets: [] };
 
 			// order of the presets is important: last preset is applied first
 			// which means the .babelrc config should look like that:
@@ -390,7 +444,7 @@ module.exports = function (log) {
 					}
 				});
 			}
-			babelConfig.presets.push(envPreset);
+			babelConfigOptions.presets.push(envPreset);
 
 			// add the presets to enable transformation of ES modules to
 			// UI5 modules and ES classes to UI5 classes
@@ -398,9 +452,9 @@ module.exports = function (log) {
 				// if the configuration option transformModulesToUI5 is an object
 				// it contains the configuration options for the plugin
 				if (typeof configuration.transformModulesToUI5 === "object") {
-					babelConfig.presets.push(["transform-ui5", configuration.transformModulesToUI5]);
+					babelConfigOptions.presets.push(["transform-ui5", configuration.transformModulesToUI5]);
 				} else {
-					babelConfig.presets.push("transform-ui5");
+					babelConfigOptions.presets.push("transform-ui5");
 				}
 			}
 
@@ -409,15 +463,15 @@ module.exports = function (log) {
 				// if the configuration option transformTypeScript is an object
 				// it contains the configuration options for the plugin
 				if (typeof configuration.transformTypeScript === "object") {
-					babelConfig.presets.push(["@babel/preset-typescript", configuration.transformTypeScript]);
+					babelConfigOptions.presets.push(["@babel/preset-typescript", configuration.transformTypeScript]);
 				} else {
-					babelConfig.presets.push("@babel/preset-typescript");
+					babelConfigOptions.presets.push("@babel/preset-typescript");
 				}
 			}
 
 			// add plugin to remove console statements
 			if (configuration?.removeConsoleStatements) {
-				babelConfig.plugins.push("transform-remove-console");
+				babelConfigOptions.plugins.push("transform-remove-console");
 			}
 
 			// add plugin to transform async statements to promises
@@ -426,7 +480,7 @@ module.exports = function (log) {
 			// not CSP compliant => therefore the Promise is the better
 			// solution than using the regenerator runtime (by default)
 			if (configuration?.transformAsyncToPromise) {
-				babelConfig.plugins.push([
+				babelConfigOptions.plugins.push([
 					"transform-async-to-promises",
 					{
 						inlineHelpers: true
@@ -439,9 +493,9 @@ module.exports = function (log) {
 			}
 
 			// include the source maps
-			babelConfig.sourceMaps = true;
+			babelConfigOptions.sourceMaps = true;
 
-			return updateBabelConfig(babelConfig);
+			return updateBabelConfigOptions(babelConfigOptions);
 		},
 
 		/**
@@ -516,9 +570,10 @@ module.exports = function (log) {
 	};
 	// expose internal functions for testing purposes
 	_this._helpers = {
-		loadBabelConfig,
-		findBabelConfig,
-		normalizeBabelConfig,
+		loadBabelConfigOptions,
+		findBabelConfigOptions,
+		normalizeBabelConfigOptions,
+		postprocessBabelConfigOptions,
 		resolvePresetOrPlugin,
 		normalizePresetOrPlugin
 	};

--- a/packages/ui5-tooling-transpile/test/util.test.js
+++ b/packages/ui5-tooling-transpile/test/util.test.js
@@ -87,28 +87,26 @@ test("inject configuration options", async (t) => {
 	let babelConfig = await t.context.util.createBabelConfig({ configuration }, cwd);
 	t.deepEqual(
 		babelConfig,
-		(
-			await t.context.util._helpers.loadBabelConfig(
-				{
-					plugins: [],
-					presets: [
-						[
-							"@babel/preset-env",
-							{
-								targets: {
-									browsers: "defaults"
-								}
+		await t.context.util._helpers.loadBabelConfigOptions(
+			{
+				plugins: [],
+				presets: [
+					[
+						"@babel/preset-env",
+						{
+							targets: {
+								browsers: "defaults"
 							}
-						],
-						"transform-ui5",
-						"@babel/preset-typescript"
+						}
 					],
-					sourceMaps: true
-				},
-				true,
-				cwd
-			)
-		).options
+					"transform-ui5",
+					"@babel/preset-typescript"
+				],
+				sourceMaps: true
+			},
+			true,
+			cwd
+		)
 	);
 
 	// typescript, custom transform-ui5 config
@@ -125,33 +123,31 @@ test("inject configuration options", async (t) => {
 	babelConfig = await t.context.util.createBabelConfig({ configuration }, cwd);
 	t.deepEqual(
 		babelConfig,
-		(
-			await t.context.util._helpers.loadBabelConfig(
-				{
-					plugins: [],
-					presets: [
-						[
-							"@babel/preset-env",
-							{
-								targets: {
-									browsers: "defaults"
-								}
+		await t.context.util._helpers.loadBabelConfigOptions(
+			{
+				plugins: [],
+				presets: [
+					[
+						"@babel/preset-env",
+						{
+							targets: {
+								browsers: "defaults"
 							}
-						],
-						[
-							"transform-ui5",
-							{
-								overridesToOverride: true
-							}
-						],
-						"@babel/preset-typescript"
+						}
 					],
-					sourceMaps: true
-				},
-				true,
-				cwd
-			)
-		).options
+					[
+						"transform-ui5",
+						{
+							overridesToOverride: true
+						}
+					],
+					"@babel/preset-typescript"
+				],
+				sourceMaps: true
+			},
+			true,
+			cwd
+		)
 	);
 
 	// typescript, custom typescript and transform-ui5 config
@@ -171,37 +167,35 @@ test("inject configuration options", async (t) => {
 	babelConfig = await t.context.util.createBabelConfig({ configuration }, cwd);
 	t.deepEqual(
 		babelConfig,
-		(
-			await t.context.util._helpers.loadBabelConfig(
-				{
-					plugins: [],
-					presets: [
-						[
-							"@babel/preset-env",
-							{
-								targets: {
-									browsers: "defaults"
-								}
+		await t.context.util._helpers.loadBabelConfigOptions(
+			{
+				plugins: [],
+				presets: [
+					[
+						"@babel/preset-env",
+						{
+							targets: {
+								browsers: "defaults"
 							}
-						],
-						[
-							"transform-ui5",
-							{
-								overridesToOverride: true
-							}
-						],
-						[
-							"@babel/preset-typescript",
-							{
-								optimizeConstEnums: true
-							}
-						]
+						}
 					],
-					sourceMaps: true
-				},
-				true,
-				cwd
-			)
-		).options
+					[
+						"transform-ui5",
+						{
+							overridesToOverride: true
+						}
+					],
+					[
+						"@babel/preset-typescript",
+						{
+							optimizeConstEnums: true
+						}
+					]
+				],
+				sourceMaps: true
+			},
+			true,
+			cwd
+		)
 	);
 });

--- a/showcases/ui5-tsapp/karma-ci-cov.conf.js
+++ b/showcases/ui5-tsapp/karma-ci-cov.conf.js
@@ -5,7 +5,7 @@ module.exports = function (config) {
 		preprocessors: {
 			//"webapp/**/!(*.d.)ts": ["ui5-transpile"],
 			"webapp/{*.ts,!(test)/**/*.ts}": ["ui5-transpile"],
-			"webapp/**/*.js": ["coverage"],
+			"webapp/{*.js,!(test)/**/*.js}": ["coverage"],
 		},
 		coverageReporter: {
 			dir: "coverage",

